src/java.base/share/classes/jdk/internal/vm/vector/VectorSupport.java

```diff
@@ -266,6 +266,20 @@ VM binaryOp(int oprId, Class<? extends VM> vmClass, Class<?> elementType, int le
         return defaultImpl.apply(vm1, vm2);
     }
 
+    @IntrinsicCandidate
+    public static
+    <V, M>
+    V binaryMaskOp(int oprId, Class<? extends V> vmClass, Class<? extends M> maskClass,
```

Comment:
Current patch make VectorSupport.binaryMaskOp the fundamental intrinsic for creating  a masked/non-masked vector nodes. Can you kindly elaborate the need for these change.

I think an ideal transformation in compiler should be able to save any changes proposed in java code, unless there is any pressing need for it.

Reply:
We proposed two solutions for generating the masked vector operations. Changing the intrinsic is one of them. Another is the same with your idea (please see the prototype: https://github.com/XiaohongGong/panama-vector/commit/372489feeae06bc53c46709d389cb0e46e9fb4f6). Actually we think both solutions can work. But they all have their advantages and shortages.